x/assets/keeper/client_chain.go

```diff
@@ -35,16 +35,52 @@ func (k Keeper) GetClientChainInfoByIndex(ctx sdk.Context, index uint64) (info *
 	return &ret, nil
 }
 
-func (k Keeper) GetAllClientChainInfo(ctx sdk.Context) (infos map[uint64]*assetstype.ClientChainInfo, err error) {
-	store := ctx.KVStore(k.storeKey)
-	iterator := sdk.KVStorePrefixIterator(store, assetstype.KeyPrefixClientChainInfo)
+// IterateAllClientChains iterates all client chains, and the `opFunc` will be called for
+// each client chain. As for the `isUpdate`, it a flag to indicate if the client chain
+// info handled by the `opFunc` will be restored.
+func (k Keeper) IterateAllClientChains(ctx sdk.Context, isUpdate bool, opFunc func(clientChain *assetstype.ClientChainInfo) error) error {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), assetstype.KeyPrefixClientChainInfo)
+	iterator := sdk.KVStorePrefixIterator(store, nil)
 	defer iterator.Close()
-
-	ret := make(map[uint64]*assetstype.ClientChainInfo, 0)
 	for ; iterator.Valid(); iterator.Next() {
```

Comment:
For my understanding, this is the part that makes it deterministic right? The iteration order is sorted in the store.

Reply:
Yeah